lib/bolt/puppetfile/module.rb

```diff
@@ -42,16 +42,39 @@ def title
       # Checks two modules for equality.
       #
       def eql?(other)
-        self.class == other.class && @owner == other.owner && @name == other.name
+        self.class == other.class &&
+          @owner == other.owner &&
+          @name == other.name &&
+          versions_intersect?(other)
```

Comment:
What do we use `eql?` for? It's weird to say that two modules with version _range_ specifications are equal just because there is overlap between their ranges. But how weird it is depends on the context in which it's used.

Reply:
It's used by `Set` to determine whether the object is already part of a set / for comparison of sets. At least that's my understanding.

Reply:
Specifically, checking if the versions intersect allows `existing_puppetfile.modules.superset? puppetfile.modules` to check if versions for each module in the Puppetfile fall within the version requirement specified in project config.

Reply:
An alternative I was looking at would remove this logic in favor of a `Bolt::Puppetfile#superset?` (or similar) method, which would compare the modules in two Puppetfiles directly, instead of relying on a comparison of sets. That would probably give some more flexibility with checking if modules in a Puppetfile conform to the version requirements without conflating the two?

Reply:
If we had a ModuleDependency class separate from Module, then something like `Bolt::Puppetfile.satisfies?([ModuleDependency, ModuleDependency, ModuleDependency])` would make sense.